java/avro/src/main/java/com/microsoft/azure/schemaregistry/kafka/avro/KafkaAvroDeserializer.java

```diff
@@ -88,7 +87,9 @@ public Object deserialize(String topic, Headers headers, byte[] bytes) {
             message.setContentType("");
         }
 
-        return this.serializer.deserializeMessageData(message, TypeReference.createInstance(Object.class));
+        return (T) this.serializer.deserializeMessageData(
```

Comment:
question on the contract with serializer here - does Type (T) needs to be implementing any specific interface (since I think all the user class always need to implement org.apache.avro.specific.SpecificRecord) and if so is there any way for our implementation here to specify that constraint? I know we can do that in C#, not sure about Java here.

Reply:
This deserializer is used for both Specific and Generic records. They each have a base class of `IndexedRecord` so the return type in the class now reflects that.